shell.go

```diff
@@ -79,11 +79,13 @@ func NewShellWithClient(url string, c *gohttp.Client) *Shell {
 			url = host
 		}
 	}
-
-	return &Shell{
-		url:     url,
-		httpcli: c,
+	var sh Shell
+	sh.url = url
```

Comment:
This doesn't respect/use `c`, which seems wrong. I should review stuff after getting first coffee.

Reply:
Or I could just pay attention instead of firing off patches...